pkg/model/multiversion/manager.go

```diff
@@ -87,25 +102,27 @@ func NewAPIVersionManager(
 
 		i, err := ackmodel.New(
 			SDKAPI,
-			apiVersion,
+			version.APIVersion,
+			metadataPath,
 			apiInfo.GeneratorConfigPath,
 			defaultConfig,
 		)
 		if err != nil {
-			return nil, fmt.Errorf("cannot create model for API version %s: %v", apiVersion, err)
+			return nil, fmt.Errorf("cannot create model for API version %s: %v", version.APIVersion, err)
 		}
-		models[apiVersion] = i
+		models[version.APIVersion] = i
 	}
 
 	sort.Strings(spokeVersions)
 	model := &APIVersionManager{
 		gitRepo:       gitRepo,
+		metadata:      metadata,
 		hubVersion:    hubVersion,
 		spokeVersions: spokeVersions,
 		apiInfos:      apisInfo,
 		models:        models,
 	}
-	// TODO(hilalymh): Audit deprecated and removed versions
```

Comment:
I'm still going to send a PR for API versions examination. Mainly to verify that service controller maintainers didn't violate one the deprecation/removal policies.